compiler/semantic/scope.go

```diff
@@ -1,18 +1,24 @@
 package semantic
 
 import (
+	"fmt"
+
+	"github.com/brimdata/zed"
 	"github.com/brimdata/zed/compiler/ast/dag"
+	"github.com/brimdata/zed/compiler/kernel"
+	"github.com/brimdata/zed/zson"
 )
 
 type Scope struct {
-	stack []Binder
+	zctx  *zed.Context
+	stack []*Binder
 }
 
 func NewScope() *Scope {
-	return &Scope{}
+	return &Scope{zctx: zed.NewContext()}
 }
 
-func (s *Scope) tos() Binder {
+func (s *Scope) tos() *Binder {
```

Comment:
`tos` is a little cryptic, so maybe call this `top`?

Reply:
I've used this forever... I think it's pretty common

https://www.acronymfinder.com/Top-of-the-Stack-(TOS).html
https://users.ece.cmu.edu/~koopman/stack_computers/sec3_2.html#:~:text=One%20of%20the%20data%20element,as%20used%20by%20the%20programmer.&text=This%20scheme%20allows%20using%20a,the%20top%20two%20stack%20elements.